src/aws.rs

```diff
@@ -0,0 +1,196 @@
+            Action::GetCredentialReport => "GetCredentialReport".to_string()
+        }
+    }
+}
+
+impl Service for AWSService {
+    fn get_users(&self) -> Result<GetUsersResult, GetUsersError> {
+        loop {
+            match self.generate_credential_report() {
+                ReportResult::COMPLETE => break,
+                _ => {}
+            };
+            sleep_ms(500); // Credential report isn't ready yet.
+        }
+
+        let aws_users = self.get_credential_report();
+
+        return Ok(GetUsersResult{
+                    service_name: "aws".to_string(),
+                    users: aws_users.iter().filter_map(|user|
+                        match user.mfa_active {
+                            false => Some(User{
+                                            name: user.user.to_string(),
+                                            email: None,
+                                            details: Some(user.arn.to_string())
+                                          }),
+                            _ => None
+                        }).collect()
+                    });
+    }
+}
+
+#[derive(Debug)]
+enum ReportResult {
+    STARTED,
+    COMPLETE,
+}
+
+impl ReportResult {
+    fn from_str(s: &str) -> ReportResult {
+        match s {
+            "STARTED" => ReportResult::STARTED,
+            "COMPLETE" => ReportResult::COMPLETE,
+            _ => panic!("Unknown report result.")
+        }
+    }
+}
+
+#[derive(RustcDecodable,Debug)]
+struct AWSUser {
+    user: String,
+    arn: String,
+    user_creation_time: String,
+    password_enabled: String,
+    password_last_used: String,
+    password_last_changed: String,
+    password_next_rotation: String,
+    mfa_active: bool,
+    access_key_1_active: String,
+    access_key_1_last_rotated: String,
+    access_key_1_last_used_date: String,
+    access_key_1_last_used_region: String,
+    access_key_1_last_used_service: String,
+    access_key_2_active: String,
+    access_key_2_last_rotated: String,
+    access_key_2_last_used_date: String,
+    access_key_2_last_used_region: String,
+    access_key_2_last_used_service: String,
+    cert_1_active: String,
+    cert_1_last_rotated: String,
+    cert_2_active: String,
+    cert_2_last_rotated: String,
+}
+
+impl AWSService {
+    fn request(&self, action: Action) -> Response {
+        let region = Region::UsEast1;
+        let mut request = SignedRequest::new(
+            "GET",
+            "iam",
+            &region,
+            "/",
+        );
+
+        request.add_param("Version", "2010-05-08");
+        request.add_param("Action", &action.as_str());
+
+        let credentials = AWSCredentials::new(
+            self.access_key_id.as_ref(),
+            self.secret_key.as_ref(),
+            None,
+            UTC::now() + Duration::seconds(600));
+
+        return request.sign_and_execute(&credentials)
+    }
+
+    fn generate_credential_report(&self) -> ReportResult {
```

Comment:
Correct, `GenerateCredentialReport` either starts a task or tells you the status of the most recent task (you get one generate credential report task every 4 hours.)